### ISSUE TITLE
all: Include create README in `lib.rs`so doctests are run

### ISSUE THREAD

**Issue by hds** [tokio-rs/tracing#2455]
### Crates

All the crates in the tracing repo (except `tracing-mock` which is already done).

### Motivation

The examples in the README files aren't tested. The README may diverge from the
top level crate documentation in `lib.rs` for each crate. There is currently a mix of
examples for current vs. future versions of each crate in the README, which may
confuse users.

### Proposal

Follow the examples in #2362 and include the full contents of the README
into the docs in `lib.rs`. As most crates already have documentation in `lib.rs`,
the two cases will need to be consolidates.

### Alternatives

We could also copy and paste the README text into `lib.rs`, but that's probably
less than ideal.

**Comment by davidbarsky** [comment]
For what it's worth, I think we should do this—it would cut down on the documentation burden tremenously.

**Comment by hawkw** [comment]
yup, i've wanted to do this for a while, but haven't had the time to work on it. if you're willing to put a PR together, that would be much appreciated!

**Comment by hds** [comment]
Yep, I've got it on my to do list. I remember we discussed this in the PR for the tracing-mock README and I kind of forgot about it for a while. But I remembered again and decided to create an issue this time so I wouldn't forget... again. (-;
